clippy_lints/src/indexing_slicing.rs

```diff
@@ -49,42 +50,34 @@ declare_clippy_lint! {
     /// Indexing and slicing can panic at runtime and there are
     /// safe alternatives.
     ///
-    /// ### Known problems
-    /// Hopefully none.
-    ///
     /// ### Example
     /// ```rust,no_run
     /// // Vector
     /// let x = vec![0; 5];
     ///
-    /// // Bad
     /// x[2];
     /// &x[2..100];
-    /// &x[2..];
-    /// &x[..100];
-    ///
-    /// // Good
-    /// x.get(2);
-    /// x.get(2..100);
-    /// x.get(2..);
-    /// x.get(..100);
     ///
     /// // Array
     /// let y = [0, 1, 2, 3];
     ///
-    /// // Bad
     /// &y[10..100];
     /// &y[10..];
-    /// &y[..100];
+    /// ```
+    ///
+    /// Use instead:
+    /// ```rust
+    /// # let x = vec![0; 5];
+    /// # let y = [0, 1, 2, 3];
+    /// # let _ =
+    /// x.get(2);
+    /// # let _ =
```

Comment:
Why are these actually needed, will we trigger a lint like `must_use` otherwise?